src/game/models/asteroid.rs

```diff
@@ -38,55 +56,92 @@ fn generate_circle(radius: f64) -> CircularPolygon {
     circle
 }
 
+/// This function takes in a shape (`CircularPolygon`) and mutates its vertices.
+/// There are few steps in this process:
+/// * add a (scaled) random amount to each dimension of each vertex
+/// * calculate the average location of the vertices. As a circle their average
+///   was [0,0], but the mutating changes this slightly
+/// * subtract the average position of the vertices from each of them. This
+///   ensures that the shape is roughly centered around 0. We aren't actually
+///   doing a real center-of-mass calculation, but this looks pretty good.
 fn randomize_shape(mut shape: CircularPolygon, max: f64) -> CircularPolygon {
-    let mut average_x = 0.0;
-    let mut average_y = 0.0;
+    // Initialize average
```

Comment:
A few of these comments are probably a little overkill (stating the obvious), but I really, really like the function doc comments. They are really useful for this part of the code.